### ISSUE TITLE
2.002 Stylistic sets in InDesign

### ISSUE THREAD

**Issue by ischaap** [psb1558/Junicode-font#236]
I've just noticed that stylistic sets all have very unhelpful names in InDesign.

Also, the stylistic sets don't seem to be working. I was trying to get at the Gothic set here, which I believe is no. 19, but it had no effect on my text, nor did any of the other stylistic sets.

![image](https://github.com/psb1558/Junicode-font/assets/139657890/c7bb5f80-c18b-4679-973f-c1ab3de7a40d)

**Comment by ischaap** [comment]
This is with the static OTF, Windows 10, InDesign 18.5.

**Comment by psb1558** [comment]
Oh God, this again. A bit of history: I discovered many months ago that InDesign messed up all Stylistic Set labels if labels for Character Variant features (even a single one) were enabled. Further, OpenType features were messed up in various other ways. I disabled the labels for the cvNN features so that the font would work in InDesign, and things continued reasonably well from there.

Until recently. I think it was @kenmcd who encouraged me to enable the cvNN labels, on the principle that it makes no sense to work around program bugs. When I did that, I found that the labels seemed no longer to be messing things up in InDesign. I celebrated with an extra glass of wine that night.

There was also a problem with InDesign crashing when the variable version was switched from roman to italic, or when the style was italic and the axis sliders were moved. This seemed to be fixed by switching from the default composer to the World-Ready Paragraph composer.

I just opened InDesign to confirm what @ischaap  was saying and discovered that not only were the labels and OT features all messed up again (in exactly the same way as before), but ID was back to crashing when the variable version was switched to italic, even when the World-Ready paragraph composer was enabled.

I've just checked through all the labels in the roman version for correctness, and I'll check the italic tomorrow.

But I don't know what else to do, and I am thinking that if the font passes all the relevant validation tests, and it is working as expected in *all* the other major programs, then I am not going to waste any more hours of my life investigating these problems.

@kenmcd, have you checked whether the current version of Affinity is displaying the cvNN and ssNN labels correctly?

**Comment by psb1558** [comment]
I just noticed that LibreOffice is now displaying both ssNN and cvNN labels (click the "Features" button in the "Character" dialog), and displaying them correctly.

**Comment by psb1558** [comment]
I just tried trashing InDesign preferences. ssNN labels/OT features still messed up, *but* no longer crashing (not immediately, anyway) with the variable font (and World-Ready composer). It may be that, with the variable font, ID is being crashed by cached info that doesn't work with an updated version of the font.

But the problem with labels/OT features remains ...

InDesign makes me so freakin' tired.

**Comment by kenmcd** [comment]
> I'd like to find other fonts with cvNN labels to test.

Charis SIL has for years.
https://github.com/silnrsi/font-charis

May be a few others in my notes.
Will check later and let you know.

**Comment by psb1558** [comment]
I can't find Source Sans 4.

Charis SIL has cv labels *and* it works correctly in InDesign. So I'm peeking inside to find out what they're doing. First thoughts:

Both Charis SIL and Junicode have format 0 cv entries.

Charis SIL is using FeatUITooltipTextNameID and SampleTextNameID, and Junicode is not. Junicode is using the Character field, and Charis SIL is not. These things shouldn't make a difference.

I was worried about whether Charis would be using Mac platform name IDs, but it doesn't (they aren't considered necessary any more, but you never know about Adobe).

I will experiment a little and report shortly.

**Comment by kenmcd** [comment]
> I can't find Source Sans 4.

Ahh, brain fart. 
Assume you found Source Sans _3._ (latest is v3.052)
https://github.com/adobe-fonts/source-sans/releases
Source _Serif 4_ does not have any cvNN features.

**Comment by kenmcd** [comment]
The Adobe OpenType Feature File Specification docs only mention the UI Name.
No mention of the Tooltip for example.
So I wonder if it is choking on the the other fields.
https://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#8d-ui-label-names-for-character-variant-features-cv01---cv99

For Affinity I had created a sample font which also added the Tooltip, the Sample Text, and the Description.
This was created in FontCreator, so it uses the format they use.
Note: this font does work fine in Affinity Publisher.
It does have multiple language support for the labels/text.
But only one character option per cvNN.
Modified version of Source Code Pro.
[SourceCodeProCV.zip](https://github.com/psb1558/Junicode-font/files/12457112/SourceCodeProCV.zip)

Wonder if InDesign has problems with the multiple character options for each cvNN.

**Comment by psb1558** [comment]
It was just the "Character" field. Omit all those, and everything worked. No problem with multiple character options (I tested Elstob with a cv that had only one). Maybe I was doing the "Charcter" field wrong, but I thought the instructions very simple and easy to follow.

I generated the extended familly and tested with that, and once again, everything worked fine. So I think we can consider this problem solved.
